spring-integration-core/src/main/java/org/springframework/integration/aggregator/ExpressionEvaluatingMessageListProcessor.java

```diff
@@ -102,7 +105,9 @@ public void setExpectedType(Class<?> expectedType) {
 	 */
 	@Override
 	public Object process(Collection<? extends Message<?>> messages) {
-		return this.evaluateExpression(this.expression, messages, this.expectedType);
+		Object object = this.evaluateExpression(this.expression, messages, this.expectedType);
+		Assert.state(object != null, "Failed to evaluate expression: " + this.expression);
```

Comment:
I don’t think it is really correct . The expression has been just evaluated to null, but we don’t expect null here. Let’s say exactly that condition: the expression evaluated to null and therefore this exception .